### PR TITLE
Apigateway authorization

### PR DESCRIPTION
**Pull request by fatima99s**

### CODE REVIEW

File: plugins/aws/apigateway/apigatewayAuthorization.js [hunk 1]
```diff
@@ -0,0 +1,69 @@
+var async = require('async');
+var helpers = require('../../../helpers/aws');
+
+module.exports = {
+    title: 'API Gateway Authorization',
+    category: 'API Gateway',
+    domain: 'Availability',
+    description: 'Ensures that Amazon API Gateway APIs are using authorizer.',
```
Comment (alphadev4) [inline plugins/aws/apigateway/apigatewayAuthorization.js:8]:
```suggestion
    description: 'Ensures that Amazon API Gateway APIs are using authorizer',
```

File: plugins/aws/apigateway/apigatewayAuthorization.js [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+                return rcb();
+            }
+
+            if (!getRestApis.data.length) {
+                helpers.addResult(results, 0, 'No API Gateway Rest APIs found', region);
+                return rcb();
+            }
+
+            getRestApis.data.forEach(api => {
+                if (!api.id) return cb();
```
Comment (alphadev4) [inline plugins/aws/apigateway/apigatewayAuthorization.js:38]: we cannot use return cb() inside the foreach loop , use retun statement instead

**Review by alphadev4 — CHANGES_REQUESTED** [review]

File: plugins/aws/apigateway/apigatewayAuthorization.js [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+                var apiArn = `arn:${awsOrGov}:apigateway:${region}::/restapis/${api.id}`;
+
+                var getAuthorizers = helpers.addSource(cache, source,
+                    ['apigateway', 'getAuthorizers', region, api.id]);
+          
+                if (!getAuthorizers || getAuthorizers.err || !getAuthorizers.data || !getAuthorizers.data.items) {
+                        helpers.addResult(results, 3,
+                            `Unable to query for API Gateway Authorizers: ${helpers.addError(getAuthorizers)}`,
+                            region, apiArn);
+                        return cb();
```
Comment (mehakseedat63) [inline plugins/aws/apigateway/apigatewayAuthorization.js:49]: this should be just return
```suggestion
                        return;
```

**Review by mehakseedat63 — CHANGES_REQUESTED** [review]
Fix linting issues

**Review by alphadev4 — APPROVED** [review]

File: plugins/aws/apigateway/apigatewayAuthorization.js [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+                var apiArn = `arn:${awsOrGov}:apigateway:${region}::/restapis/${api.id}`;
+
+                var getAuthorizers = helpers.addSource(cache, source,
+                    ['apigateway', 'getAuthorizers', region, api.id]);
+          
+                if (!getAuthorizers || getAuthorizers.err || !getAuthorizers.data || !getAuthorizers.data.items) {
+                    helpers.addResult(results, 3,
+                        `Unable to query for API Gateway Authorizers: ${helpers.addError(getAuthorizers)}`,
+                        region, apiArn);
+                    return ;
```
Comment (mehakseedat63) [inline plugins/aws/apigateway/apigatewayAuthorization.js:49]:
```suggestion
                    return;
```

**Review by mehakseedat63 — APPROVED** [review]

**Pull request merged**